src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -843,7 +855,16 @@ private int decodeProtocolHeaderZero(
             final int protocolId = protocolHeader.id();
             server.decoder = decodePlainFrame;
 
-            server.onDecodeProtocolHeader(traceId, authorization, protocolHeader);
+            connectionState = connectionState.receivedHeader();
+            if (connectionState != ERROR)
+            {
+                server.onDecodeProtocolHeader(traceId, authorization, protocolHeader);
+            }
+            else
+            {
+                server.onDecodeError(traceId, authorization, ILLEGAL_STATE, null);
+            }
+
```

Comment:
Recommend you invert the logic, using a `decode:` label.
```
decode:
if (protocolHeader != null)
{
    connectionState = connectionState.receivedHeader();
    if (connectionState == ERROR)
    {
        server.onDecodeError(traceId, authorization, ILLEGAL_STATE, null);
        break decode;
    }

    final int protocolId = protocolHeader.id();
    server.decoder = decodePlainFrame;
    server.onDecodeProtocolHeader(traceId, authorization, protocolHeader);
```

Also, should `server.onDecodeError` be `decodeError` instead to change `decoder` to `decodeIgnoreAll`?

This feedback applies to all places where we detect `ERROR` connection state.